pgx/src/array.rs

```diff
@@ -1,23 +1,104 @@
 use crate::datum::{Array, FromDatum};
-use crate::pg_sys::{self, bits8, ArrayType};
+use crate::pg_sys;
 use bitvec::prelude::*;
 use bitvec::ptr::{bitslice_from_raw_parts_mut, BitPtr, BitPtrError, Mut};
 use core::ptr::{slice_from_raw_parts_mut, NonNull};
 use core::slice;
 
-extern "C" {
-    /// # Safety
-    /// Does a field access, but doesn't deref out of bounds of ArrayType
-    fn pgx_ARR_DATA_PTR(arrayType: *mut ArrayType) -> *mut u8;
-    /// # Safety
-    /// Does a field access, but doesn't deref out of bounds of ArrayType
-    fn pgx_ARR_DIMS(arrayType: *mut ArrayType) -> *mut libc::c_int;
-    /// # Safety
-    /// Must only be used on a "valid" (Postgres-constructed) ArrayType
-    fn pgx_ARR_NELEMS(arrayType: *mut ArrayType) -> i32;
-    /// # Safety
-    /// Does a field access, but doesn't deref out of bounds of ArrayType
-    fn pgx_ARR_NULLBITMAP(arrayType: *mut ArrayType) -> *mut bits8;
```

Comment:
The safety comments got lost. The ported functions should still document, mostly, whether they're a deref or only an offset computation (and thus whether the field accessed must be properly initialized, or whether we merely must be pointing at a large enough palloc). These are fairly obvious from reading the code but it's useful to have it registered in the description of the function which will be previewed by tooling, no need for an extensive remark.